tests/test_rss.py

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2016 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#     Santiago Dueñas <sduenas@bitergia.com>
+#     Alvaro del Castillo <acs@bitergia.com>
+#
+
+import argparse
+import json
+import shutil
+import sys
+import tempfile
+import unittest
+
+import httpretty
+import pkg_resources
+
+# Hack to make sure that tests import the right packages
+# due to setuptools behaviour
+sys.path.insert(0, '..')
+pkg_resources.declare_namespace('perceval.backends')
+
+from perceval.cache import Cache
+from perceval.errors import CacheError
+from perceval.backends.core.rss import (RSS,
+                                            RSSCommand,
+                                            RSSClient)
+
+
+RSS_FEED_URL = 'http://example.com/rss'
+
+requests_http = []
+
+def read_file(filename, mode='r'):
+    with open(filename, mode) as f:
+        content = f.read()
+    return content
+
+class TestRSSBackend(unittest.TestCase):
+    """RSS backend tests"""
+
+    def test_initialization(self):
+        """Test whether attributes are initializated"""
+
+        rss = RSS(RSS_FEED_URL, tag='test')
+
+        self.assertEqual(rss.url, RSS_FEED_URL)
+        self.assertEqual(rss.origin, RSS_FEED_URL)
+        self.assertEqual(rss.tag, 'test')
+        self.assertIsInstance(rss.client, RSSClient)
+
+        # When tag is empty or None it will be set to
+        # the value in url
+        rss = RSS(RSS_FEED_URL)
+        self.assertEqual(rss.url, RSS_FEED_URL)
+        self.assertEqual(rss.origin, RSS_FEED_URL)
+        self.assertEqual(rss.tag, RSS_FEED_URL)
+
+        rss = RSS(RSS_FEED_URL, tag='')
+        self.assertEqual(rss.url, RSS_FEED_URL)
+        self.assertEqual(rss.origin, RSS_FEED_URL)
+        self.assertEqual(rss.tag, RSS_FEED_URL)
+
+    def test_has_caching(self):
+        """Test if it returns True when has_caching is called"""
+
+        self.assertEqual(RSS.has_caching(), True)
+
+    def test_has_resuming(self):
+        """Test if it returns False when has_resuming is called"""
+
+        self.assertEqual(RSS.has_resuming(), False)
+
+    def __configure_http_server(self):
```

Comment:
Yes, it was used only inside this class this is why I left it there but I have moved it outside the class.